tests/python/pants_test/engine/test_scheduler.py

```diff
@@ -224,15 +238,19 @@ def test_get_type_match_failure(self):
 
   def test_unhashable_failure(self):
     """Test that unhashable Get(...) params result in a structured error."""
-    expected_msg = """TypeError: unhashable type: 'list'"""
-    with assert_execution_error(self, expected_msg):
+    with assert_execution_error(self, dedent("""\
+      TypeError: For object with type 'list', error was: unhashable type: 'list'""")):
       self.scheduler.product_request(C, [Params(TypeCheckFailWrapper(A()))])
 
+    with assert_execution_error(self, dedent("""\
+      TypeError: For object with type 'CollectionType', error was: unhashable type: 'list'""")):
```

Comment:
I don't suppose it's easy/possible to add the field name here too, is it?

Reply:
We probably could, by modifying the `__hash__` of `datatype`. I will look into this since that is a clear usability improvement.

Reply:
Done -- was able to remove the change in `native.py` as a result!